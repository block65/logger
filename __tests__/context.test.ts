import {
  afterEach,
  beforeEach,
  describe,
  expect,
  jest,
  test,
} from '@jest/globals';

describe('Context Wrapper', () => {
  const initialEnv = process.env;

  beforeEach(() => {
    jest.useFakeTimers({ now: new Date('2009-02-13T23:31:30.000Z') });
  });

  afterEach(() => {
    process.env = { ...initialEnv };
    jest.useRealTimers();
  });

  test('Lambda', async () => {
    const { withLambdaLoggerContextWrapper } = await import('../lib/lambda.js');

    const { createAutoConfiguredLoggerWithWaitableMock } = await import(
      './helpers.js'
    );

    const [logger, callback, errback] =
      createAutoConfiguredLoggerWithWaitableMock();

    logger.info('logger outside before');

    const childLogger = logger.child(
      {
        childData: '1',
      },
      {
        context: {
          childCtx: '1',
        },
      },
    );

    await withLambdaLoggerContextWrapper(
      childLogger,
      {
        awsRequestId: 'fake-request-id1',
        functionVersion: '999',
      },
      async () => {
        logger.info('logger inside');
        childLogger.info('childLogger inside');
      },
    );

    logger.info('logger outside after');

    await logger.end();

    await expect(callback.waitUntilCalledTimes(4)).resolves.toMatchSnapshot();

    expect(errback).not.toBeCalled();
  });
});
