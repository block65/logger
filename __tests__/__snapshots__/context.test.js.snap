// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context Wrapper Basic 1`] = `
Array [
  Array [
    Object {
      "ctx": Object {
        "name": "CONTEXTNAME",
      },
      "level": 30,
      "msg": "hello",
      "time": 2009-02-13T23:31:30.000Z,
    },
  ],
]
`;

exports[`Context Wrapper Lambda 1`] = `
Array [
  Array [
    "2009-02-13T23:31:30.000Z		INFO	logger outside before	{\\"level\\":30}
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z		INFO	logger inside	{\\"level\\":30}
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z		INFO	logger outside after	{\\"level\\":30}
",
  ],
  Array [
    "2009-02-13T23:31:30.000Z		INFO	childLogger inside	{\\"level\\":30,\\"childData\\":\\"1\\"}
",
  ],
]
`;
